libgnucash/engine/Account.cpp

```diff
@@ -2752,23 +2752,34 @@ DxaccAccountSetCurrency (Account * acc, gnc_commodity * currency)
 /********************************************************************\
 \********************************************************************/
 
+void
+gnc_account_foreach_descendant (const Account *acc, std::function<void(Account*)> account_cb)
+{
+    g_return_if_fail (GNC_IS_ACCOUNT(acc));
+
+    // some callers eg xaccAccountTreeScrubLots will modify the children
+    auto children = GET_PRIVATE(acc)->children;
```

Comment:
Sadly if a reference is created instead of a vector copy, test-lots will fail.

Reply:
That shouldn't surprise you.